uavcan/node/435.ExecuteCommand.0.1.uavcan

```diff
@@ -0,0 +1,68 @@
+# The parameter contains the path to the new software image file to be downloaded by the server from the client
+# using the standard service uavcan.file.Read. Observe that this operation swaps the roles of the client and
+# the server. Upon reception of this command, the server (updatee) will evaluate whether it is possible to
+# begin the software update process. If that is deemed impossible, the command will be rejected with one of the
+# error codes defined in the response section of this definition (e.g., DENIED if the node is currently on-duty
+# and a sudden interruption of its activities is considered unsafe, and so on). If an update process is already
+# underway, the updatee should abort the process and restart with the new file, unless the updatee can
+# determine that the specified file is the same file that is already being downloaded, in which case it is
+# allowed to respond SUCCESS and continue the old update process. If there are no other conditions precluding
+# the requested update, the updatee will return a SUCCESS and initiate the file transfer process by invoking
+# the standard service uavcan.file.Read repeatedly until the file is transferred fully (please refer to the
+# documentation for that data type for more information on its usage). While the software is being updated, the
+# updatee should set its mode (the field "mode" in uavcan.node.Status) to MODE_SOFTWARE_UPDATE. Please refer to
+# the documentation for uavcan.node.Status for more information.
+uint16 COMMAND_BEGIN_SOFTWARE_UPDATE = 65533
+
+# Return the node's configuration back to the factory default settings (may require restart).
+# Due to the uncertainty whether a restart is required, generic interfaces should always force a restart.
+uint16 COMMAND_FACTORY_RESET = 65532
+
+# Cease activities immediately, enter a safe state until restarted.
+# Further operation may no longer be possible until a restart command is executed.
+uint16 COMMAND_EMERGENCY_STOP = 65531
+
+# This is related to "uavcan.register".
+# This command instructs the node to store the current configuration parameter values and other persistent states
+# to the non-volatile storage. Nodes are allowed to manage persistent states automatically, obviating the need for
+# this command by committing all such data to the non-volatile memory automatically as necessary. However, some
+# nodes may lack this functionality, in which case this parameter should be used. Generic interfaces should always
+# invoke this command in order to ensure that the data is stored even if the node doesn't implement automatic
+# persistence management.
+uint16 COMMAND_STORE_PERSISTENT_STATES = 65530
+
+# A string parameter supplied to the command. The format and interpretation is command-specific.
+# The standard commands do not use this field (ignore it), excepting the following:
+#   - COMMAND_BEGIN_SOFTWARE_UPDATE
+uint8[<=200] parameter
```

Comment:
I'd call this "payload" but I'm not very good at naming things, apparently.

Reply:
The objective here is not to transfer some data to the server, but to enrich the command with additional data that specifies its behavior. In that sense, "payload" doesn't work well, but "argument" or "data" do.

Reply:
Okay. I don't feel too strongly about this. parameter is fine. My thinking with "payload" is that sometimes (perhaps most of the time) this will be a parameter but sometimes it will be data. "Payload" is more of an generic name for bits that can be either.